### PR TITLE
cleanup(docs): update release.md

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

> /area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

- Updates and minor refinements to reflect stable modern_bpf driver support starting with Falco release >= 0.35.
- Additional minor updates and adjustments to improve clarity.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
cleanup(docs): update release.md
```

### CODE REVIEW

File: falco.yaml [hunk 1]
```diff
@@ -903,7 +903,7 @@ base_syscalls:
   custom_set: []
   repair: false
 
-# [Experimental] `modern_bpf.cpus_for_each_syscall_buffer`, modern_bpf only
+# [Stable] `modern_bpf.cpus_for_each_syscall_buffer`, modern_bpf only
```
Comment (incertum) [inline falco.yaml:906]: @Andreagit97 correcting mistake of wrong maturity level when I adjusted falco.yaml style and formatting.

File: RELEASE.md [hunk 2]
```diff
@@ -205,13 +212,13 @@ Announce the new release to the world!
...
+- [FALCO_ENGINE_VERSION](https://github.com/falcosecurity/falco/blob/master/userspace/engine/falco_engine_version.h) is not sem-ver and must be bumped either when a backward incompatible change has been introduced to the rules files syntax or `falco --list -N | sha256sum` has changed. Breaking changes introduced in the Falco engine are not necessarily tied to the drivers or libs versions. The primary idea behind the hash is that when new filter / display fields (see currently supported [Falco fields](https://falco.org/docs/rules/supported-fields/)) are introduced a version bump indicates that this field was not available in previous engine versions. See the [rules release guidelines](https://github.com/falcosecurity/rules/blob/main/RELEASE.md#versioning-a-ruleset) to understand how this affects the versioning of Falco rules. Lastly, `FALCO_ENGINE_VERSION` is incremented only once during a Falco release cycle.
```
Comment (incertum) [inline RELEASE.md:220]: added a necessary clarification here.

File: RELEASE.md [hunk 3]
```diff
@@ -52,28 +57,30 @@ The Falco project publishes all drivers for each release for all popular kernel
...
+Moreover, we assign owners for each release (typically pairing a new person with an experienced one). Assignees and due dates for releases are proposed during the [weekly community call](https://github.com/falcosecurity/community).
 
 At a high level each Falco release needs to follow a pre-determined sequencing of releases and build order:
 
 - [1 - 3] `libs` (+ `driver`) and `plugins` components releases
 - [4] Falco driver pre-compiled object files push to Falco's Artifacts repo
 - [5] Falco userspace binary release
 
-Finally, on the proposed due date the assignees for the upcoming release proceed with the processes described below.  
+Assignees are responsible for creating a Falco GitHub issue to track the release tasks and monitor the progress of the release. This issue serves as a central point for communication and provides updates on the release dates. You can refer to the [Falco v0.35 release](https://github.com/falcosecurity/falco/issues/2554) or [Libs Release (0.11.0+5.0.1+driver)](https://github.com/falcosecurity/libs/issues/1092) issues as examples/templates for creating the release issue.
```
Comment (incertum) [inline RELEASE.md:77]: new addition

File: RELEASE.md [hunk 4]
```diff
@@ -5,18 +5,22 @@
...
+> Note: Starting with Falco 0.35.x releases, the Falco userspace binary includes the `modern_bpf` driver object code during the linking process. This integration is made possible by the CO-RE (Compile Once - Run Everywhere) feature of the modern BPF driver. CO-RE allows the driver to function on kernels that have backported BTF (BPF Type Format) support or have a kernel version >= 5.8. For the older `kmod` and `bpf` drivers, separate artifacts are released for the kernel space. This is because these drivers need to be explicitly compiled for the specific kernel release, using the exact kernel headers. This approach ensures that Falco can support a wide range of environments, including multiple kernel versions, distributions, and architectures.  (see `libs` [driver - kernel version support matrix](https://github.com/falcosecurity/libs#drivers-officially-supported-architectures)).
```
Comment (incertum) [inline RELEASE.md:15]: @Andreagit97 would you have additional preferences or ok?

**Comment by FedeDP** [comment]
I'll put this in the 0.35.0 milestone; if we run out of time, since this is just a documentation PR, we can move it to 0.36.0!
/milestone 0.35.0

**Review by leogr — DISMISSED** [review]
LGTM. Thank you!

Just left a comment (not a blocker, anyway).

**Comment by FedeDP** [comment]
I compeletly forgot to publish the review :/ sorry @incertum ahaha

**Review by leogr — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**